include/ppconsul/kv.h

```diff
@@ -329,15 +332,95 @@ namespace ppconsul { namespace kv {
         std::string m_defaultDc;
     };
 
-    // TODO: add ScopedLock
-
-    // Implementation
-
     namespace impl {
         StringList parseKeys(const std::string& resp);
         std::vector<KeyValue> parseValues(const std::string& resp);
+
+        namespace session {
+            std::string createBodyJson(const std::string &node, int lockDelay,
+                                       const std::string &behavior, int ttl);
+            std::string parseCreateResponse(const std::string &resp);
+        }
     }
 
+    class Sessions
+    {
+    public:
+        enum class OnInvalidation
+        {
+            RELEASE,
+            DELETE,
+        };
+
+        struct DestroyError: public ppconsul::Error
```

Comment:
Why do you need to have a special exception for a mailformed delete session response?

Reply:
Should I just ignore the result?

Reply:
`return parseJsonBool(...)` should be fine. The caller can decide what to do in case of false result (probably nothing).

Reply:
The fun is, it returns true even if called with non-existing session ID.